### ISSUE TITLE
Missing :authority header 

### ISSUE THREAD

**Issue by johndpope** [nathanborror/swift-grpc#3]
Using this 
https://github.com/johndpope/swift3-grpc-example
to test against
https://github.com/johndpope/DockerParseyMcParsefaceAPI

am seeing **"Missing :authority header"**

<img width="824" alt="screen shot 2016-10-22 at 12 41 43" src="https://cloud.githubusercontent.com/assets/289994/19621459/70ec5e86-9860-11e6-874a-50ea846ce5b6.png">

When I run the node client against server - it correctly returns data.
https://github.com/johndpope/DockerParseyMcParsefaceAPI/blob/master/clients/node_client/index.js

Re-looking at the node sample it appears obvious that an 'insecure'/unauthenticated credential is created to login.
`var service = new protoDescriptor.cali.nlp.ParseyService("0.0.0.0:9000", grpc.credentials.createInsecure());`

I'll keep digging.

http://www.grpc.io/grpc/node/src_credentials.js.html

/**
- Create an insecure credentials object. This is used to create a channel that
- does not use SSL. This cannot be composed with anything.
- @return {ChannelCredentials} The insecure credentials object
  */
  exports.createInsecure = ChannelCredentials.createInsecure;

```

 from http_server_filter.c

static void hs_on_recv(grpc_exec_ctx *exec_ctx, void *user_data, bool success) {
  grpc_call_element *elem = user_data;
  call_data *calld = elem->call_data;
  if (success) {
    server_filter_args a;
    a.elem = elem;
    a.exec_ctx = exec_ctx;
    grpc_metadata_batch_filter(calld->recv_initial_metadata, server_filter, &a);
    /* Have we seen the required http2 transport headers?
       (:method, :scheme, content-type, with :path and :authority covered
       at the channel level right now) */
    if (calld->seen_method && calld->seen_scheme && calld->seen_te_trailers &&
        calld->seen_path && calld->seen_authority) {
      /* do nothing */
    } else {
      if (!calld->seen_path) {
        gpr_log(GPR_ERROR, "Missing :path header");
      }
      if (!calld->seen_authority) {
        gpr_log(GPR_ERROR, "Missing :authority header");
      }
      if (!calld->seen_method) {
        gpr_log(GPR_ERROR, "Missing :method header");
      }
      if (!calld->seen_scheme) {
        gpr_log(GPR_ERROR, "Missing :scheme header");
      }
      if (!calld->seen_te_trailers) {
        gpr_log(GPR_ERROR, "Missing te trailers header");
      }
      /* Error this call out */
      success = 0;
      grpc_call_element_send_cancel(exec_ctx, elem);
    }
  }
  calld->on_done_recv->cb(exec_ctx, calld->on_done_recv->cb_arg, success);
}
```

```


        var headers: [(HeaderField, String)] = [
            (.method, "POST"),
            (.scheme, url.scheme ?? "http"),
            (.path, path),
            (.contentType, "application/grpc+proto"),
            (.te, "trailers"),
            (.authority, "www.test.com"),
           // (.userAgent, "Swift-http-client/2.0")
            ]


seems to get passed the error - but now fails silently.

```

**Comment by johndpope** [comment]
so I looked around - and re-checked files
it seems protoc doesn't automatically generate the 'service' component in swift.
It obviously does generate the message stubs.

fyi - I found this project - well worth reviewing.
https://github.com/hosopy/grpc-mobile-demo-ios

I noticed the convention is to initialise class using 
        GRPCCall.useInsecureConnectionsForHost("localhost:50051")

Also - within this code base - there's objective-c code that has the service / but it's not clear whether a protoc --plugin generated this or was done by hand. 
I think this is why my service is not returning anything.

These are third party plugins available to generate code
https://github.com/google/protobuf/blob/master/docs/third_party.md
